genai-perf/genai_perf/config/generate/genai_perf_config.py

```diff
@@ -56,3 +59,37 @@ def _set_options_based_on_objective(
                         self.input.__setattr__(
                             name, parameter.get_value_based_on_category()
                         )
+
+    ###########################################################################
+    # Checkpoint Methods
+    ###########################################################################
+    def write_to_checkpoint(self) -> Dict[str, Any]:
+        """
+        Converts the class data into a dictionary that can be written to
+        the checkpoint file
+        """
+        genai_perf_config_dict = deepcopy(self.__dict__)
+
+        return genai_perf_config_dict
+
+    @classmethod
+    def read_from_checkpoint(
+        cls, genai_perf_config_dict: Dict[str, Any]
+    ) -> "GenAIPerfConfig":
+        """
+        Takes the checkpoint's representation of the class and creates (and populates)
+        a new instance of a GenAIPerfConfig
+        """
+        genai_perf_config = GenAIPerfConfig(
+            config=ConfigCommand([""]),
```

Comment:
Does it make sense to store the command that generated a run in the checkpoint?

Reply:
This is how (in future runs/subcommands) we will identify if the results have already been obtained.

Reply:
I misunderstood this piece of code.
Discussed offline and this is actually really well designed.